packages/checkout/widgets-lib/src/widgets/sale/SaleWidget.tsx

```diff
@@ -54,14 +62,33 @@ export function SaleWidget(props: SaleWidgetProps) {
     eventTargetState: { eventTarget },
   } = useContext(EventTargetContext);
 
+  const env = checkout!.config.environment ?? Environment.SANDBOX;
+
   const { theme } = config;
   const biomeTheme = useMemo(() => widgetTheme(theme), [theme]);
 
   const [viewState, viewDispatch] = useReducer(viewReducer, initialViewState);
-  const viewReducerValues = useMemo(() => ({ viewState, viewDispatch }), [viewState, viewDispatch]);
+  const viewReducerValues = useMemo(
+    () => ({ viewState, viewDispatch }),
+    [viewState, viewDispatch],
+  );
+  const { currencyResponse: currency } = useCurrency({
+    env,
+    environmentId,
+    currencyName: CURRENCY_NAME,
+  });
+
+  const [fromTokenAddress, setFromTokenAddress] = useState('');
 
-  const loadingText = viewState.view.data?.loadingText
-    || t('views.LOADING_VIEW.text');
+  const loadingText = viewState.view.data?.loadingText || t('views.LOADING_VIEW.text');
+
+  useEffect(() => {
```

Comment:
@mimi-imtbl setting a state again is not really necessary. as you can destructure currency values here
```  const { erc20Address: fromTokenAddress } = useCurrency({
    env,
    environmentId,
    currencyName: CURRENCY_NAME,
  });```